### ISSUE TITLE
Map page is blank when router changed from other router

### ISSUE THREAD

**Issue by haies** [ghettovoice/vuelayers#102]
router is
``` html
  {
    path: '/',
    component: Layout,
    redirect: '/dashboard',
    name: 'Dashboard',
    hidden: true,
    children: [{
      path: 'dashboard',
      component: () => import('@/views/dashboard/index')
    }]
  },

  {
    path: '/resource/show',
    component: Layout,
    meta: { keepAlive: true },
    children: [
      {
        path: '',
        component: () => import('@/views/resource-show/index'),
        name: 'resourceShow',
        meta: { title: 'resourceShow', icon: 'example', noCache: true }
      }
    ]
  }
```
page structure is
```html
  <div :class="classObj" class="app-wrapper">
    <div v-if="device==='mobile'&&sidebar.opened" class="drawer-bg" @click="handleClickOutside"/>
    <sidebar class="sidebar-container"/>
    <div class="main-container">
      <navbar/>
      <app-main/>
    </div>
  </div>
```
The map is nested in app-main, and it is
```html
  <section class="app-main">
    <transition name="fade-transform" mode="out-in">
      <keep-alive>
        <router-view v-if="$route.meta.keepAlive"/>
      </keep-alive>
    </transition>
    <transition name="fade-transform" mode="out-in">
      <router-view  v-if="!$route.meta.keepAlive"/>
    </transition>
  </section>
</template>
``` 
The "/resource/show" page is blank when router is changed from “/dashbord”, but it display rightly when I refresh the chrome or adjust the size of the window of chrome

**Comment by ghettovoice** [comment]
Hi @haies, 
can you confirm that canvas in the map has width and height equal to 0 after route change?

**Comment by haies** [comment]
> Hi @haies,
> can you confirm that canvas in the map has width and height equal to 0 after route change?

the map canvas width and height are not equal to 0 , and they are the right value

**Comment by ghettovoice** [comment]
Thank you for the repo :+1: 

I checked it and found that this is css related issue. In the last page, where map is not ok, the resource container `.resource-show-container` doesn't have any size at render time. While it's parent has dynamically defined height with `min-height`, resource show container should have an absolute height because `vl-map` container trying to fill the parent container with `height: 100%`. But percent height doesn't work inside parent with min-height.

If you for example stretch r`.resource-show-container` then all work as expected.

```css
.resource-show-container {
    position: absolute;
    bottom: 0;
    top: 0;
    left: 0;
    right: 0;
  }
```

In other words the `vl-map` should be always rendered inside container with defined rect size. This is also true if you use native openlayers

**Comment by haies** [comment]
> Thank you for the repo 👍
> 
> I checked it and found that this is css related issue. In the last page, where map is not ok, the resource container `.resource-show-container` doesn't have any size at render time. While it's parent has dynamically defined height with `min-height`, resource show container should have an absolute height because `vl-map` container trying to fill the parent container with `height: 100%`. But percent height doesn't work inside parent with min-height.
> 
> If you for example stretch r`.resource-show-container` then all work as expected.
> 
> ```css
> .resource-show-container {
>     position: absolute;
>     bottom: 0;
>     top: 0;
>     left: 0;
>     right: 0;
>   }
> ```
> 
> In other words the `vl-map` should be always rendered inside container with defined rect size. This is also true if you use native openlayers. Strictly speaking OpenLayers uses canvas size at the initialization to correctly calculate current view resolution

I see light suddenly，this question is bothered me quiet a long time, thank you again. @ghettovoice 👍
